external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala

```diff
@@ -79,4 +80,23 @@ class AvroOptions(
   val compression: String = {
     parameters.get("compression").getOrElse(SQLConf.get.avroCompressionCodec)
   }
+
+  @transient private val acceptableParseMode = Seq(PermissiveMode, FailFastMode)
```

Comment:
do we need to check mode here? The `AvroOptions` can be used with reading avro files, and DROP MALFORMED can be supported.